tests/cpp/test_allocation_domain.cpp

```diff
@@ -1423,4 +1423,19 @@ TEST_F(AllocationDomainTest, ReductionVectorization) {
   testValidate(executor_cache.fusion(), cg_outputs, inputs, __LINE__, __FILE__);
 }
 
+TEST_F(AllocationDomainTest, ClearReductionIterDomainsPatch) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  auto tv0 = TensorViewBuilder()
+                 .ndims(3)
+                 .shape({-1, 1, -1})
+                 .contiguity({true, std::nullopt, true})
+                 .build();
+  auto tv1 = sum(tv0, {0});
+  tv1->setAllocationDomain(
+      {tv1->axis(0), tv1->axis(2), tv1->axis(1)},
+      {std::nullopt, true, std::nullopt});
+  tv1->clearReductionIterDomains();
```

Comment:
Shouldn't this do some sort of validation?

Reply:
will do